src/compas/datastructures/mesh/duality.py

```diff
@@ -24,24 +23,41 @@ def mesh_dual(mesh, cls=None, boundary=0):
         Defaults to the type of the provided mesh object.
     boundary: float, optional
         boundary mode for the dual mesh
-        Default mode is 0, not create faces on boundaries
+        Default mode is 0, not create faces on boundaries.
+        1, create faces on mesh edges, not include original mesh boundary vertices.
+        2. create faces on mesh edges and include original mesh boundary vertices on the corner.
+        3. create faces on mesh edges and include all original mesh boundary vertices.
 
     Returns
     -------
     :class:`~compas.datastructures.Mesh`
         The dual mesh object.
 
+    Examples
+    --------
+    >>> from compas.datastructures import Mesh
+    >>> mesh = Mesh.from_obj(compas.get('faces.obj'))
+    >>> mesh.delete_face(11)
+    >>> mesh.delete_face(6)
+    >>> mesh.delete_face(7)
+    >>> mesh.quads_to_triangles()
+    >>> mesh = mesh.subdivide('corner')
+    >>> dual = mesh.dual(boundary=3)
+
     """
     if not cls:
         cls = type(mesh)
 
     dual = cls()
 
+    mesh.unify_cycles()
```

Comment:
potentially expensive so should be left up to the caller

Reply:
I will remove it.